### ISSUE TITLE
Add a way to clear all events in a calendar

### ISSUE THREAD

**Issue by eesko** [owncloud/calendar#326]
Currently there seems to be no way of clearing all events in a calendar other than to do it manually one event at a time.

Previously one could tick an option to clear all events in a calendar when importing events. This option does not exist anymore.

**Comment by georgehrke** [comment]
@jancborchardt

**Comment by jancborchardt** [comment]
Um … just delete the calendar before? ;)

**Comment by brot** [comment]
Deleting the calendar before is no option.
I use CalDav on my phone. Deleting the calendar before and creating a new one with the imported data means, that I have to reconfigure my phone. But I'm using a shared calendar, so everyone using this shared calendar on the phone needs to configure the new created calendar.

So the easiest way to resolve this issue is to bring back the option on import to delete the calendar before importing the data. 
This option was there before (Owncloud calendar <9) and should be here with the new calendar.

**Comment by aspdye** [comment]
Why is the reconfiguration needed? The new calendar will have the same name, the same CalDAV URLs and credintals :wink: you only need to reconfigure the shares, or am i wrong :grin:

**Comment by jancborchardt** [comment]
Exactly, there is no re-configuration needed when the name stays the same.

**Comment by brot** [comment]
Maybe, but reconfiguring somethin is wrong and useless. What If you add a typo when creating the new calendar? And reconfiguring the shares?! This would be an extra step for every import.
With the old version of the calendar app this wasn't necessary. 

I have to import an external calendar every few weeks. And therefore I need to delete all the entries of the calendar before importing the new events. This calendar is shared with a few people. 

So, what's the problem with adding the option to delete all entries before importing the new one? Are there any technical problems?

**Comment by georgehrke** [comment]
> I have to import an external calendar every few weeks. 

Maybe the webcal feature is what you are waiting for? 
https://github.com/owncloud/calendar/issues/132

**Comment by georgehrke** [comment]
> Are there any technical problems?

It clutters the user interface and it leaves us with more code to maintain.

**Comment by brot** [comment]
> Maybe the webcal feature is what you are waiting for?
> #132

I'm not sure at the moment. I think the university only provides the ics file as a download. So this wouldn't work

> It clutters the user interface and it leaves us with more code to maintain.

This would be one more ui element and one cleanup routine before importing the calendar

**Comment by georgehrke** [comment]
>  ics file as a download

ICS files on the web is essentially what webcal is

> This would be one more ui element and one cleanup routine before importing the calendar

But it's used by very very very few users and if we start implementing every little feature that would benefit 0.0001% of our users the calendar would become a real usability pain.

**Comment by brot** [comment]
> But it's used by very very very few users and if we start implementing every little feature that would benefit 0.0001% of our users the calendar would become a real usability pain.

Hard to tell how much user would use this feature. At the moment there is @eesko who created this issue and me who would like to see this feature back in the calendar too.
And maybe a lot more out there who are still using the old calendar version or don't know anything about issues on github. 
But yes, in the end it's the decision of the ownCloud maintainers.

**Comment by brot** [comment]
> ICS files on the web is essentially what webcal is

Yes, let me guess. Configuring webcal behind user login would also be too much to maintain.
That's the feature I would need to use webcal, because I have to login before I can see/download the ICS file. 
Sadly there is not much information on #132, so I can't see if this feature would be included

**Comment by georgehrke** [comment]
Well, if you can access the the ics file by sending http basic auth headers it will definitely work

**Comment by ege1** [comment]
I'm also missing the "clear calendar" feature. While it wasn't really intuitive before (uploading the ICS file first, to import), it was convenient as I was able to just drag and drop the ICS file and import it, replacing the whole calendar.
It is also OK for me to delete the calendar first, but less convenient....

**Comment by georgehrke** [comment]
@ege1 What do you use it for? Deleting all objects in a calendar and starting all over again on a regular basis doesn't make sense in my opinion.

**Comment by georgehrke** [comment]
>  it was convenient as I was able to just drag and drop the ICS file

That's indeed missing, i created an issue for it: https://github.com/owncloud/calendar/issues/468

**Comment by ege1** [comment]
@georgehrke Yes, it sounds a bit weird. I'm periodically syncing my personal calendar at my employers system (by exporting ICS file) to my private owncloud calendar, to be able to see business schedules on my private phone.

**Comment by eesko** [comment]
There are some issues with just deleteng and recreating a calendar with the same name. See: #325

**Comment by georgehrke** [comment]
Instead of reimplementing an old crappy feature you used to solve your problem we should think about different, more comfortable, more intuitive ways to solve your problem.

**Comment by jancborchardt** [comment]
Yeah, isn’t this exactly what subscribing to remote calendars would be for?

**Comment by georgehrke** [comment]
If it's a plain ics file on the web (maybe behind basic auth) yes,
If it's a different CalDAV/Exchange server, then no.

**Comment by ege1** [comment]
@georgehrke: I'm with you. It is surely not a normal use-case that I'm using here and if I would really rely on it, I would be searching for a proper/automated solution.
But I wasn't aware that the old feature was crappy. For me it worked flawless.

**Comment by brot** [comment]
So, at the moment the feature to delete all events from a calendar is missing. Also the webcal support (#132) is missing, but is the preferred solution to solve this issue.
When should we be able to install a version from the Calendar app (1.2?) which includes this feature? At the moment I have to weekly delete an calendar, import the new ics file and reassign all the shared users.

**Comment by hitam4450** [comment]
I certainly agree, that the current version 1.1 of the calendar app in OC 9.0.1 is far away from  being usable properly!  :thumbsdown: :thumbsdown: :thumbsdown:

**Comment by XuRuHang** [comment]
I do fully agree on this point. Since the upgrade to OC 9 I really miss two features. I export my professional Lotus Notes agenda by generating a .ics file. Then I import it into my OC calendar because I have there other personal events. In return, my OC calendar is synchronized with my Android devices.
The 2 features I miss since OC 9 :
1) delete all events from calendar before importing ics file
2) be able to automate cleaning calendar and importing ics files.

In the time being, is there some workaround by deleting some entries in a database table ?

There is also a problem in OC 9 : I cannot open any calendar entry when i click on it after import. Unable to see details then.

**Comment by georgehrke** [comment]
> 2) be able to automate cleaning calendar and importing ics files.

We never had this feature.

**Comment by XuRuHang** [comment]
@georgehrke ooops yes that's correct. I was using your cl-calendarimport https://github.com/georgehrke/cl-calendarimport but now it is not working on OC 9 ("won't fix") :-(

**Comment by DeepDiver1975** [comment]
Instead of crying for features and telling us how badly this is needed: contribute!

You know: this is open source!

**Comment by XuRuHang** [comment]
Sorry but everyone here knows that's open source. Thanks. In my opinion, it's good there are feedback from users, even if in this point some complained because of a kind of regression in a feature they liked, because it points out a feature that is needed by your **precious** users.
Most of us are not coders (or good coders in the language of the application which is my case).
But anyway if I can help by describing my use case better, feel free to ask.

**Comment by georgehrke** [comment]
@XuRuHang And you can't sync your Lotus Notes agenda directly with your android? 
Apparently you don't share it with other users. What's the benefit of piping it thru ownCloud?

**Comment by XuRuHang** [comment]
@georgehrke 
The main reason is that I manage also personal events on my owncloud calendar and I don't want to enter these events on my professional Notes agenda. My OC agenda handles then all events both professional and personal.
My android device is easy to configure to be synchronized with OC calendars too.

**Comment by h1-ffm** [comment]
Hallo 
I, am one of the  very very very few users and need the feature also.

**Comment by caco3** [comment]
As shown https://github.com/owncloud/calendar/issues/132#issuecomment-227876626, there is a need to wipe a calendar also for me.

**Comment by EtlamGit** [comment]
I also occasionally have to clear all events from a calendar. When something goes wrong it is a convenient way to import a backup for a calendar without all the configuration for sharing options or even syncing on clients.

**Comment by gohrner** [comment]
Apparently, issue #804 was a duplicate of this. In #804 it was mentioned, that sharing webcal subscriptions is planned (https://github.com/nextcloud/server/issues/1600, not yet finished), which will probably help in several cases.

However, I also have the case that someone sends me an ICS file from time to time which I need to import and which is not publicly available on the web, so I also need the "clear existing calendar and import" feature if I do not want to nag my users about needing to re-enable a re-created calendar...
